### PR TITLE
chore(layouts): 12585, 12580, 12588, 12590, 12607 - implement dynamic layouts

### PR DESCRIPTION
**Pull request by IlyaIzr**
1920
image
Laptop
image
Mobile
image
image
image

### CODE REVIEW

**Comment by Komzpa** [comment]
screenshots LGTM. the fitBounds behavior is not shown on the screenshots.

File: src/views/Main/Layouts/Desktop/Desktop.module.css [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+  gap: var(--unit);
+  padding: var(--unit);
+
+  /* analytics+layers column, advanced analytics column, map column, layers&legends column */
+  grid-template-columns:
+    minmax(320px, 1fr) fit-content(400px) minmax(440px, 10fr)
+    minmax(320px, 1fr);
+  grid-template-rows: calc(100% - 40px) 40px;
+}
+/* .firstColumn, .secondColumn, .thirdColumn, .fourthColumn{
```
Comment (propakov) [inline src/views/Main/Layouts/Desktop/Desktop.module.css]: why to save this comment? can't find it in code or comments

**Review by propakov — COMMENTED** [review]
From code side it LGTM, but I've found some bugs while testing.
1) resizing corners are a bit different
image
2) scroll bar on mobile view goes out of screen, and it's not possible to scroll to the bottom of the list
near the bottom
image
the most bottom position
image

File: src/core/store/columnContext.tsx [hunk 2]
```diff
@@ -0,0 +1,121 @@
+import { useContext, createContext } from 'react';
+import type { MutableRefObject } from 'react';
+
+export const ColumnContext = createContext<Resizer | string>('no provider was given');
```
Comment (propakov) [inline src/core/store/columnContext.tsx]: I also can't find a reason to have string there. We don't put there other strings, only one place here. 
Maybe just null? And there `if (typeof columnContext === 'string') return;` process as null

Comment (IlyaIzr) [reply src/core/store/columnContext.tsx]: i was hoping it would give better readability, but since you've asked, it didn't - so i'll change it

**Comment by propakov** [comment]
@IlyaIzr after fix it's double scroll 
Screenshot 2022-10-06 at 16 18 43
and icon on mobile is without shadows
image

File: src/core/store/columnContext.tsx [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+    resizeObserver.observe(this.column.current);
+    this.destroy = () => resizeObserver.unobserve(this.column!.current!);
+  }
+
+  addPanel(panel: PanelMeta) {
+    return this.panels.add(panel);
+  }
+
+  destroy() {
+    // TODO
```
Comment (Akiyamka) [inline src/core/store/columnContext.tsx]: what you want todo here?

File: src/views/Main/Main.tsx [hunk 4]
```diff
@@ -1,20 +1,19 @@
...
 import type { UserDataModel } from '~core/auth';
-import type { MutableRefObject } from 'react';
 
-const { EditFeaturesOrLayerPanel } = lazily(
-  () =>
-    import(
-      '~features/create_layer/components/EditFeaturesOrLayerPanel/EditFeaturesOrLayerPanel'
-    ),
-);
+// To be restored later
```
Comment (Akiyamka) [inline src/views/Main/Main.tsx]: Please add some info about when in must be restored

**Comment by Akiyamka** [comment]
Nice work!

**Review by Akiyamka — APPROVED** [review]
Code looks nice, just few comments about comments : )

**Pull request merged**